nemo/collections/asr/data/audio_to_label.py

```diff
@@ -162,10 +165,10 @@ def sliced_seq_collate_fn(self, batch):
                 assumes the signals are 1d torch tensors (i.e. mono audio).
             fixed_length (Optional[int]): length of input signal to be considered
         """
-        slice_length = self.featurizer.sample_rate * self.time_length
+        slice_length = int(self.featurizer.sample_rate * self.time_length)
         _, audio_lengths, _, tokens_lengths = zip(*batch)
         slice_length = min(slice_length, max(audio_lengths))
-        shift = 1 * self.featurizer.sample_rate
+        shift = int(self.featurizer.sample_rate * self.shift_length)
         has_audio = audio_lengths[0] is not None
 
         audio_signal, num_slices, tokens, audio_lengths = [], [], [], []
```

Comment:
Yes good idea. Basically I just 'ignore' last slice and user prediction of the previous slice.  Since in overlapped smoothing filter it will automatically be replaced therefore fixed. Thanks for the suggestion, as your above comment. Use if condition (if slices <= 0:) and append zeros for it. Will update.

Reply:
Do not update if slices<=0 section, I repeat the remaining audio to time_length if input_audio is less than time_length. You might only need to change else section, but in vad_infer.py make a note in file description that to users to make sure input audio file is greater than time_length

Reply:
Why not do the same thing - repeat the remaining audio to time_length if input_audio is less than time_length for the last slice in VAD?

Reply:
oh the forward infer loop in vad_infer.py

Reply:
You could do that too, anything but just need to add something to the last missing slice in else section to make end output vad length == audio length